src/components/Button.js

```diff
@@ -1,16 +1,21 @@
 import React, { Component } from 'react';
-
 import PropTypes from "prop-types";
 import Big from "big.js";
+import calculate from '../logic/calculate';
 
 export default class Button extends Component {
   constructor(props) {
     super(props)
   };
 
   render() {
+    const data = this.props.data;
```

Comment:
* [ ] You need to validate all your props with `props-type` package. This is important for debugging especially when the application becomes bigger and bigger.

Reply:
I have added another prop type validation in the Button component, checking that `this.props.processInput` is present and is a function.